File: src/java/magmac/app/lang/node/Catch.java
package magmac.app.lang.node;

import magmac.api.Option;
import magmac.app.compile.error.CompileResult;
import magmac.app.compile.node.Node;
import magmac.app.lang.Destructors;
import magmac.app.lang.java.JavaDefinition;

public record Catch(JavaDefinition definition) implements JavaBlockHeader {
    public static Option<CompileResult<JavaBlockHeader>> deserialize(Node node) {
        return Destructors.destructWithType("catch", node).map(deserializer -> deserializer.withNode("definition", JavaDefinition::deserialize)
                .complete(Catch::new));
    }
}

File: src/web/magmac/app/lang/node/Catch.ts
import { Option } from "../../../../magmac/api/Option";
import { CompileResult } from "../../../../magmac/app/compile/error/CompileResult";
import { Node } from "../../../../magmac/app/compile/node/Node";
import { Destructors } from "../../../../magmac/app/lang/Destructors";
import { JavaDefinition } from "../../../../magmac/app/lang/java/JavaDefinition";
export class Catch {
	deserialize(node : Node) : Option<CompileResult<JavaBlockHeader>> {return 0;;}
}
